packages/scripts/utils/config.js

```diff
@@ -363,16 +351,15 @@ function getWebpackEntryPoints( buildType ) {
 			cwd: fromProjectRoot( getProjectSourcePath() ),
 		} );
 
-		if ( ! entryFile ) {
+		if ( entryFile ) {
+			entryPoints.index = entryFile;
```

Comment:
Merge the `index` entry with any prior entries. Alternatively, we could skip this if custom entry points are present, assuming that users have specified all custom files they wanted.